modules/packages/Python.chpl

```diff
@@ -2800,6 +2800,66 @@ module Python {
     proc get(idx: int): owned Value throws do
       return this.get(owned Value, idx);
 
+    /*
+      Get a slice of the list indicated by ``bounds``.  Equivalent to
+      calling ``obj[bounds.low:bounds.high+1]`` in Python.
+
+      .. note::
+
+         This method does not support strided ranges or ranges with an alignment
+         other than 0.
+
+      :arg T: the Chapel type of the slice to return.
+      :arg bounds: The full slice of the list to return
+      :returns: A slice of the list for the given bounds.
+    */
+    pragma "docs only"
+    proc get(type T = owned Value, bounds: range(?)): T throws do
+      compilerError("docs only");
+
+    @chpldoc.nodoc
+    proc get(type T, bounds: range(?)): T throws {
+      if (bounds.strides != strideKind.one) {
+        compilerError("cannot call `get()` on a Python list with a range with stride other than 1");
+      }
+
+      var ctx = chpl_pythonContext.enter();
+      defer ctx.exit();
+
+      var pyObj;
+
+      if (bounds.hasLowBound() && bounds.hasHighBound()) {
+        pyObj = PyList_GetSlice(this.getPyObject(),
+                                bounds.low.safeCast(Py_ssize_t),
+                                bounds.high.safeCast(Py_ssize_t) + 1);
+
+      } else if (!bounds.hasLowBound() && bounds.hasHighBound()) {
+        pyObj = PyList_GetSlice(this.getPyObject(), min(Py_ssize_t),
+                                bounds.high.safeCast(Py_ssize_t) + 1);
+
+      } else if (bounds.hasLowBound() && !bounds.hasHighBound()) {
+        pyObj = PyList_GetSlice(this.getPyObject(),
+                                bounds.low.safeCast(Py_ssize_t),
+                                max(Py_ssize_t));
+
+      } else {
+        pyObj = PyList_GetSlice(this.getPyObject(), min(Py_ssize_t),
+                                max(Py_ssize_t));
+      }
+      this.interpreter.checkException();
+      return interpreter.fromPythonInner(T, pyObj);
```

Comment:
given that this logic is now duplicated many times, do you think its worthwhile to factor it out into a helper function that returns a tuple, `(low, high)`, that is unpacked and then used in `Py*_*Slice`?